rust/src/web/graphql/blocks/mod.rs

```diff
@@ -131,11 +131,14 @@ impl BlocksQueryRoot {
         }
 
         // global slot query
-        if let Some(global_slot_since_genesis) =
-            query.as_ref().and_then(|q| q.global_slot_since_genesis)
+        if let Some(global_slot_since_genesis) = query
+            .as_ref()
+            .and_then(|q| q.protocol_state.as_ref())
```

Comment:
Should this be `.or()`? Otherwise the query must provide them all

Reply:
@Isaac-DeFrain If you are certain, then you can push a commit with a change. Otherwise, you'll have to merge this PR unless it is malformed.